app/src/main/java/com/dessalines/thumbkey/db/AppDb.kt

```diff
@@ -613,8 +635,41 @@ val MIGRATION_16_17 =
         }
     }
 
+val MIGRATION_17_18 =
+    object : Migration(17, 18) {
+        override fun migrate(db: SupportSQLiteDatabase) {
+            db.execSQL(
+                "ALTER TABLE AppSettings ADD COLUMN auto_size_keys INTEGER NOT NULL DEFAULT $DEFAULT_AUTO_SIZE_KEYS",
+            )
+            // The default for this new setting is *enabled*, but that would be a change for existing users, so if
+            // migrating from a previous dabase version withouth the setting at all, assume the user manually set
+            // the actual size they want to use, hence defaulting legacy users to *disabled* on upgrade.
+            db.execSQL(
+                "UPDATE AppSettings SET auto_size_keys = 0",
+            )
+            db.execSQL(
+                "ALTER TABLE AppSettings ADD COLUMN non_square_keys INTEGER NOT NULL DEFAULT $DEFAULT_NON_SQUARE_KEYS",
+            )
+            db.execSQL(
+                "UPDATE AppSettings SET non_square_keys = 1 WHERE key_width IS NOT NULL",
+            )
+            db.execSQL(
+                "ALTER TABLE AppSettings RENAME COLUMN key_size TO key_height",
+            )
+            db.execSQL(
+                "ALTER TABLE AppSettings MODIFY COLUMN key_height INTEGER NOT NULL default $DEFAULT_KEY_HEIGHT",
+            )
+            db.execSQL(
+                "UPDATE AppSettings SET key_width = $DEFAULT_KEY_WIDTH WHERE key_width IS NULL",
+            )
+            db.execSQL(
+                "ALTER TABLE AppSettings MODIFY COLUMN key_width INTEGER NOT NULL default $DEFAULT_KEY_WIDTH",
```

Comment:
I'm sure this one isn't going to work, you need to make a new field and rename the old one to defunct.